l10n_br_account_banking_payment_cnab/model/payment_order.py

```diff
@@ -37,7 +38,76 @@ class PaymentOrder(models.Model):
+            raise Warning("Company's Rezão Social should not be longer than 30 chars")
+        if not self.mode.boleto_protesto:
+            raise Warning(u"Códigos de Protesto in payment mode not defined")
+        if not self.mode.boleto_protesto_prazo:
+            raise Warning(u"Prazo protesto in payment mode not defined")
+        else:
+            try:
+                int(self.mode.boleto_protesto_prazo)
+            except:
+                raise Warning("Prazo protesto in payment mode must be integer")
+        if not self.mode.bank_id.bra_number_dig:
+            raise Warning("Dígito Agência not defined")
+        else:
+            try:
+                int(self.mode.bank_id.bra_number_dig)
+            except:
+                raise Warning("Dígito Agência must be integer")
 
+            
+        # move lines must have transaction refernce
+        for line in self.line_ids:
+            if not line.partner_id:
+                raise Warning("Partner not defined for %s" %line.name)
+            if not line.partner_id.legal_name:
+                raise Warning("Rezão Social not defined for %s" %line.partner_id.name)
+            if len(line.partner_id.legal_name) > 30:
+                raise Warning("Partner's Rezão Social should not be longer than 30 chars")
+            if not line.partner_id.state_id:
+                raise Warning("Partner's state not defined")
+            if not line.partner_id.state_id.code:
+                raise Warning("Partner's state code not defined")
+            # max 15 chars
+            if not line.partner_id.district:
+                raise Warning("Partner's bairro not defined")
+            else:
+                if len(line.partner_id.district) > 15:
+                    raise Warning("Partner's bairro %s can't be more than 15 digits" %line.partner_id.district)
+            if not line.partner_id.zip:
+                raise Warning("Partner's CEP not defined")
+            if not line.partner_id.l10n_br_city_id:
+                raise Warning("Partner's city not defined")
+            if not line.partner_id.street:
+                raise Warning("Partner's street not defined")
+            if not line.move_line_id.transaction_ref:
+                raise Warning("No transaction reference set for move %s" % line.move_line_id.name)
+            if len(line.move_line_id.transaction_ref) > 8:
+                raise Warning("Transaction reference for move line can not be longer than 8 digits found %s" %line.move_line_id.transaction_ref)
+            try:
+                int(line.move_line_id.transaction_ref)
+            except:
+                raise Warning("Transaction reference for move line must be integer")
+            if not line.move_line_id.invoice.number:
+                raise Warning("Null value in 'numero_documento' number not defined for invoice %s" % line.move_line_id.invoice.number)
+            if len(line.move_line_id.invoice.number) > 10:
```

Comment:
This was considered only for itaú where we consider transaction ref with 8 digits = sequence of boleto